README.md

```diff
@@ -132,6 +136,31 @@ params:
   - compute.googleapis.com/instance/disk
 ```
 
+### What to know about Aggregating DELTA Metrics
+
+Treating DELTA Metrics as a gauge produces data which is wildly inaccurate/not very useful (see https://github.com/prometheus-community/stackdriver_exporter/issues/116). However, aggregating the DELTA metrics overtime is not a perfect solution and is intended to produce data which mirrors GCP's data as close as possible. 
+
+The biggest challenge to producing a correct result is that a counter for prometheus does not start at 0, it starts at the first value which is exported. This can cause inconsistencies when the exporter first starts and for slow moving metrics which are described below.
+
+#### Start-up Delay
+
+When the exporter first starts it has no persisted counter information and the stores will be empty. When the first sample is received for a series it is intended to be a change from a previous value according to GCP, a delta. But the prometheus counter is not initialized to 0 so it does not export this as a change from 0, it exports that the counter started at the sample value. Since the series exported are dynamic it's not possible to export an [initial 0 value](https://prometheus.io/docs/practices/instrumentation/#avoid-missing-metrics) in order to account for this issue. The end result is that it can take a few cycles for aggregated metrics to start showing rates exactly as GCP. 
+
+As an example consider a prometheus query, `sum by(backend_target_name) (rate(stackdriver_https_lb_rule_loadbalancing_googleapis_com_https_request_bytes_count[1m]))` which is aggregating 5 series. All 5 series will need to have two samples from GCP in order for the query to produce the same result as GCP.
+
+#### Slow Moving Metrics
+
+A slow moving metric would be a metric which is not constantly changing with every sample from GCP. GCP does not consistently report slow moving metrics DELTA metrics. If this occurs for too long (default 5m) prometheus will mark the series as [stale](https://prometheus.io/docs/prometheus/latest/querying/basics/#staleness). The end result is that the next reported sample will be treated as the start of a new series and not an increment from the previous value. Here's an example of this in action, ![](https://user-images.githubusercontent.com/4571540/184961445-ed40237b-108e-4177-9d06-aafe61f92430.png)
+
+There are two features which attempt to combat this issue, 
+
+1. `monitoring.aggregate-deltas-ttl` which controls how long a metric is persisted in the data store after its no longer being reported by GCP
+1. Metrics which were not collected during a scrape are still exported at their current counter value
+
+The configuration when using `monitoring.aggregate-deltas` gives a 30 minute buffer to slower moving metrics and `monitoring.aggregate-deltas-ttl` can be adjusted to tune memory requirements vs correctness. Storing the data for longer results in a higher memory cost.
```

Comment:
I wonder if we should follow the Prometheus default and have the staleness TTL be 5 minutes.

Reply:
I ended up going for a higher default because there are a reasonable amount of metrics with a note of `After sampling, data is not visible for up to x seconds` where x > 5 minutes. 30 minutes might be a bit overzealous/arbitrary so I don't mind dropping it down further. I will say so far it hasn't seemed to cause a large increase in memory usage in our environment.

Reply:
Ok, makes sense. I didn't know Stackdriver would provide such low granularity for some things. The 30m default is fine.